src/ray/core_worker/core_worker.cc

```diff
@@ -190,13 +194,22 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
 
   task_manager_.reset(new TaskManager(
       memory_store_, reference_counter_, actor_manager_,
-      [this](const TaskSpecification &spec) {
+      /*retry_task_callback=*/[this](const TaskSpecification &spec) {
         // Retry after a delay to emulate the existing Raylet reconstruction
         // behaviour. TODO(ekl) backoff exponentially.
         RAY_LOG(ERROR) << "Will resubmit task after a 5 second delay: "
                        << spec.DebugString();
+        const auto now = current_time_ms();
         absl::MutexLock lock(&mutex_);
-        to_resubmit_.push_back(std::make_pair(current_time_ms() + 5000, spec));
+        if (spec.IsActorTask()) {
+          const auto actor_id = spec.ActorId();
+          // Schedule a kill task unless we already have one in flight.
+          if (!pending_kill_for_actor_.contains(actor_id)) {
+            pending_actor_kills_.emplace_back(now + kActorRestartDeadlineMillis, spec.ActorId());
+            pending_kill_for_actor_[actor_id] = std::prev(pending_actor_kills_.end());
+          }
```

Comment:
I don't think we should be killing the actor during the retry process, is this needed?

Reply:
The concern here is about whether we should allow clients to kill the actor. I think we should probably just fail all of the pending tasks on this client instead of killing the actor outright (which could end up failing tasks for other clients that did not receive a transient error).

Reply:
Yeah I guess the killing might not be necessary in this version of the patch. I can change it to keep trying to resubmit the tasks indefinitely.